analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Invocation.cs

```diff
@@ -172,10 +172,23 @@ private static ProgramState ProcessAssertedBoolSymbol(ProgramState state, IOpera
 
     private static ProgramState[] ProcessEquals(SymbolicContext context, IInvocationOperationWrapper invocation)
     {
-        if (invocation.TargetMethod.IsStatic && invocation.Arguments.Length == 2
-            && invocation.Arguments[0].ToArgument().Value is var leftOperation
-            && invocation.Arguments[1].ToArgument().Value is var rightOperation
-            && context.State[leftOperation]?.Constraint<ObjectConstraint>() is var leftConstraint
+        if (invocation.Arguments.Length == 2 && invocation.TargetMethod.IsStatic)
+        {
+            return ProcessEquals(context, invocation.Arguments[0].ToArgument().Value, invocation.Arguments[1].ToArgument().Value);
+        }
+        else if (invocation.Arguments.Length == 1 && invocation.TargetMethod.ContainingType.IsNullableValueType())
```

Comment:
What about if the argument `IsNullableValueType`? Is this the same conceptually?
```cs
int i1 = 0;
int? i2  = null;
i1.Equals(i2); // Target: NonNullable, Arg: Nullable
i2.Equals(i1); // Target: Nullable, Arg: NonNullable
```

Reply:
Consider adding a link to the spec where the null rules for `==` are described and that somehow motivate this special handling:
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/nullable-value-types#lifted-operators
If this isn't the motivation, I could argue, that `someObject.Equals(someKnownNull)` always returns `false` for all types.

Reply:
While `any.Equals(null)` _should_ always return `false`, it does not have to.

This is supposed to add support for `Nullable<T>.Equals` method, because we know how that specific one behaves.
We could do something similar for a bunch of others, like `int.Equals`, but that's out of scope here. We might add it when adding the CBDE literal constraint.

Reply:
Makes sense. I would add a link to the documentation for `Nullable<T>.Equals()` to show that this is well-defined behavior.

Reply:
I wouldn't add it. The switch expression is pretty explicit about the nullable and there's nothing weird going on.